File: ruby/Prototype/prototype_pattern.rb
baseCar = Car.new
baseCar.name = "Focus S"
baseCar.printDetails

print "\n"
otherVersion  = (Car) PrototypeFactory.getInstance(baseCar);
otherVersion.name = "Focus SE"
otherVersion.AutomaticClimateControl = true
otherVersion.CruiseControl = true
otherVersion.printDetails
           

File: java/Prototype/src/PrototypePattern.java
public class PrototypePattern
{
    public static void main(String[] args)
    {
        try
        {
        	Car baseCar = new Car();
        	baseCar.setName("Focus S");
        	baseCar.printDetails();
        	
        	System.out.println("\n");
            Car otherVersion  = (Car) PrototypeFactory.getInstance(baseCar);
            otherVersion.setName("Focus SE");
            otherVersion.setAutomaticClimateControl(true);
            otherVersion.setCruiseControl(true);
            otherVersion.printDetails();
           
 
        }
        catch (CloneNotSupportedException e)
        {
        	System.out.println("Object not cloneable");
            e.printStackTrace();
        }
    }
}